### ISSUE TITLE
Yume2kki: Game crashes in certain places

### ISSUE THREAD

**Issue by coinsandboins828626** [EasyRPG/Player#2593]
Thank you for supporting EasyRPG Player.
Before creating an issue, please verify that your bug still occurs in the "continuous builds" of the Player.
You can get them at the bottom of our download section: https://easyrpg.org/player/downloads

---
Please fill in the following fields before submitting an issue:
#### Name of the game:

Yume 2kki (version 0.116)

#### Player platform:

Android

#### Attach files (as a .zip archive or link them)

[Save01.zip](https://github.com/EasyRPG/Player/files/6857821/Save01.zip)
[easyrpg_log[1].txt](https://github.com/EasyRPG/Player/files/6857826/easyrpg_log.1.txt)

#### Describe the issue in detail and how to reproduce it:

Specific to Android, Yume 2kki will crash in a few places while playing. The first is when entering Urotsuki's Dream Apartments from the black door in the Nexus, or entering that specific map from any other location. The next is when visiting the Witch's Room in Jigsaw Puzzle World. Finally, there is one more that occurs deeper in the game, when interacting with an NPC that would teleport you somewhere. Usually it goes like this: You enter another map, the new music will play for a bit with the screen still black, and then the game will crash without any error message whatsoever, going back to the game selection screen. The DreamApartments one is probably the easiest to reproduce, just sleep, go to the Nexus, and enter the black door and it should crash. It hasn't always been like this, though, I remember some versions ago it worked just fine, and I had no troubles. However, this was just for the Dream Apartments, I'm unsure how long the other two have been around.

**Comment by purupurupuruin** [comment]
IIRC I had the same problem with it crashing on Android just before entering the Vending Machine Factory, to make things easier I made a debug save just before the Vending Machine Factory, one before the Dream Apartments, and because I didn't know if by Witch's Room you meant the room with the witch in it or the room the witch was guarding I made saves just before both lol.

[potentially crashes on android 2kki saves.zip](https://github.com/EasyRPG/Player/files/6858520/potentially.crashes.on.android.2kki.saves.zip)

**Comment by Ghabry** [comment]
This doesn't crash on Linux but UBSAN reports:

```
game_player.cpp:754:24: runtime error: shift exponent 53 is too large for 32-bit type 'int'
```

**Comment by Ghabry** [comment]
Yeah this crashes on a 32bit Android (armeabi-v7a) but survives on aarch64

**Comment by Ghabry** [comment]
Simplest solution is replacing ``2 << speed`` with ``2 * (speed + 1)``. This gets rid of the overflow and will simply accept this crap.

**Comment by Ghabry** [comment]
Sorry my solution is nonsense, the value is also totally wrong xD

The problem here is that this overflows the integer, so this should be sanitized.

**Comment by Ghabry** [comment]
I will add a temporary patch that clamps the values but keep this here open and rename it to remind us to that the RPG_RT behaviour must be checked.

**Comment by Ghabry** [comment]
This crash affected 7 users 20 times in the last 60 days on Android (according to crash reporting, ARM64 does not crash which would explain why almost nobody noticed this)
